Indexing/ClauseVariantIndex.hpp

```diff
@@ -51,6 +51,7 @@ class ClauseVariantIndex
 
 class SubstitutionTreeClauseVariantIndex : public ClauseVariantIndex
 {
+  using LiteralSubstitutionTree = Indexing::LiteralSubstitutionTree<LiteralClause>;
```

Comment:
Not the point of the change, but remind me why a *clause* variant index keeps a literal in it?

Reply:
Well I never touched it, except for doing refactoring that should make everything do what it used to do, but looking into it briefly just now revealed that it only uses a substitution tree to match the "main literal" (`getMainLiteral`) of a clause and then does matches the rest of the literals on a per-result basis. I think this miiiight be a performance bottlekneck in some cases but I can't tell how much this really happens in practice.